sklearn/metrics/tests/test_regression.py

```diff
@@ -566,7 +566,7 @@ def test_mean_pinball_loss_on_constant_predictions(distribution, target_quantile
         # Check that the loss of this constant predictor is greater or equal
         # than the loss of using the optimal quantile (up to machine
         # precision):
-        assert pbl >= best_pbl - np.finfo(best_pbl.dtype).eps
+        assert pbl >= best_pbl - np.finfo(type(best_pbl)).eps
```

Comment:
Let's be explicit about the expectation:

```suggestion
        assert pbl >= best_pbl - np.finfo(np.float64).eps
```

Reply:
Just out of curiosity:

I remember you made a similar comment when we worked on a different PR. Why do we prefer `np.float64` from NumPy over Python’s built-in `float` in these cases? From the perspective of the Array API or scikit-learn, how can I tell the difference between the two?